### PR TITLE
HttpService Rework + Trellis and Stargazer Messages and Config

### PR DESCRIPTION
**Pull request by mbolt35**
Most of this work aims to fix the issues with HttpService and multiple services. See reviews for various dependencies:

* https://github.com/enklu/commons-gradle/pull/3
* https://github.com/enklu/commons-unity-http/pull/8

### CODE REVIEW

**Comment by mbolt35** [comment]
Sorry for all of the merge noise. The specific commit of mine is here: https://github.com/enklu/enkluplayer/pull/278/commits/a41fa136caa47e6410a48dbb15f805d50c12ec6e

File: Assets/Resources/ApplicationConfig.json [hunk 1]
```diff
@@ -26,14 +26,28 @@
...
+                "Name": "stage",
+                "TrellisUrl": "https://cloud-beta.enklu.com:10001/v1",
+                "StargazerUrl": "https://cloud-beta.enklu.com:9973/v1",
+                "MyceliumIp": "54.245.63.209",
+                "MyceliumPort": 10103,
+                "AssetsUrl": "https://s3-us-west-2.amazonaws.com:443/stage.assets.bundles/",
+                "BundlesUrl": "https://s3-us-west-2.amazonaws.com:443/stage.assets.bundles/",
+                "ThumbsUrl": "https://s3-us-west-2.amazonaws.com:443/stage.assets.thumbs/",
+                "AnchorsUrl": "https://s3-us-west-2.amazonaws.com:443/stage.anchors/",
+                "ScriptsUrl": "https://s3-us-west-2.amazonaws.com:443/stage.scripts/"
```
Comment (thegoldenmule) [inline Assets/Resources/ApplicationConfig.json]: Remove!

File: Assets/Source/Application/Injection/EnkluPlayerModule.cs [hunk 2]
```diff
@@ -108,6 +109,9 @@ public void Load(InjectionBinder binder)
                 binder.Bind<HttpRequestCacher>().To<HttpRequestCacher>().ToSingleton();
                 binder.Bind<AwsPingController>().To<AwsPingController>().ToSingleton();
                 binder.Bind<ApiController>().To<ApiController>().ToSingleton();
+                binder.Bind<CreateAR.Stargazer.Messages.ApiController>()
+                      .To<CreateAR.Stargazer.Messages.ApiController>()
+                      .ToSingleton();
```
Comment (thegoldenmule) [inline Assets/Source/Application/Injection/EnkluPlayerModule.cs]: COOOOL

File: Assets/Source/Player/Scripting/Interface/App/SessionJsApi.cs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+        /// <summary>
+        /// Dispatched when a session is created.
+        /// </summary>
+        private const string EVENT_SESSION_CREATED = "created";
+
+        /// <summary>
+        /// Dispatched when a session ends.
+        /// </summary>
+        private const string EVENT_SESSION_ENDED = "ended";
```
Comment (thegoldenmule) [inline Assets/Source/Player/Scripting/Interface/App/SessionJsApi.cs]: These could be public, though it's a broader question about how our JS APIs should work. Would we rather have:

`sessions.on(session.EVENT_SESSION_ENDED, onEvent)`

or

`sessions.on('ended', onEvent);`

We can actually add autocomplete to the former.

File: Assets/Source/Player/Session/IPlayerSessionController.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+        /// <summary>
+        /// Creates a new session.
+        /// </summary>
+        IAsyncToken<PlayerSession> CreateSession();
+
+        /// <summary>
+        /// Ends the current session.
+        /// </summary>
+        void EndSession();
```
Comment (thegoldenmule) [inline Assets/Source/Player/Session/IPlayerSessionController.cs]: From an interface level, I assume `EndSession` synchronously fails in progress `CreateSession` tokens? If so, I think it should be in the comment.

File: Assets/Source/Player/Session/IPlayerSessionController.cs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+        /// <summary>
+        /// The current player session.
+        /// </summary>
+        PlayerSession CurrentSession { get; }
+
+        /// <summary>
+        /// Creates a new session.
+        /// </summary>
+        IAsyncToken<PlayerSession> CreateSession();
```
Comment (thegoldenmule) [inline Assets/Source/Player/Session/IPlayerSessionController.cs]: What happens when you call `CreateSession` while another session is in progress? That should probably described as part of the interface and should be commented here.

File: Assets/Source/Player/Session/SessionQrViewController.cs [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        [InjectElements("..img-qr")]
+        public ImageWidget Qr { get; set; }
+        [InjectElements("..caption-progress")]
+        public TextWidget Progress { get; set; }
+
+        /// <inheritdoc />
+        protected override void AfterElementsCreated()
+        {
+            base.AfterElementsCreated();
+
```
Comment (thegoldenmule) [inline Assets/Source/Player/Session/SessionQrViewController.cs]: Empty line -- in fact this whole function can be removed

File: Assets/Source/Player/Session/SessionQrViewController.cs [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        /// <summary>
+        /// Shows a message instead of the QR image.
+        /// </summary>
+        /// <param name="message">The message to show.</param>
+        public void ShowMessage(string message)
+        {
+            Qr.Schema.Set("visible", false);
+
+            Progress.Schema.Set("label", message);
+            Progress.Schema.Set("visible", true);
```
Comment (thegoldenmule) [inline Assets/Source/Player/Session/SessionQrViewController.cs]: This totally works. I believe there are also shortcuts to both of these: `Progress.Label` and `Progress.LocalVisible`.

File: Assets/Source/Player/Session/StargazerCredentials.cs [hunk 8]
```diff
@@ -0,0 +1,40 @@
+﻿using CreateAR.Commons.Unity.Http;
+
+namespace CreateAR.EnkluPlayer.Player.Session
+{
+
+    public class StargazerCredentials
```
Comment (thegoldenmule) [inline Assets/Source/Player/Session/StargazerCredentials.cs]: Random empty line and no missing class comment.

File: Assets/Source/Player/Scripting/Interface/App/SessionJsApi.cs (hunk 3 above)
Comment (mbolt35) [reply Assets/Source/Player/Scripting/Interface/App/SessionJsApi.cs]: I'm 100% for the former, but we don't seem to use that anywhere else?

File: Assets/Source/Player/Session/IPlayerSessionController.cs (hunk 4 above)
Comment (mbolt35) [reply Assets/Source/Player/Session/IPlayerSessionController.cs]: Fair

File: Assets/Source/Player/Session/SessionQrViewController.cs (hunk 6 above)
Comment (mbolt35) [reply Assets/Source/Player/Session/SessionQrViewController.cs]: Yeah, I left it there in the event I wanted to add something, but never did :)

File: Assets/Source/Player/Scripting/Interface/App/SessionJsApi.cs (hunk 3 above)
Comment (thegoldenmule) [reply Assets/Source/Player/Scripting/Interface/App/SessionJsApi.cs]: It's not very javascripty.

**Pull request merged**